first_validation.py

```diff
@@ -48,11 +50,23 @@ def check_hashes(hash_fname):
     # Directory containing hash filenames file.
     data_dir = hash_pth.parent
     # Read in text for hash filename
+    text_file = hash_pth.read_text()
     # Split into lines.
+    lines = text_file.splitlines()
     # For each line:
+    for line in lines:
         # Split each line into expected_hash and filename
+        a = line.split('  ')
         # Calculate actual hash for given filename.
+        path = data_dir / a[1]
+        content = path.read_bytes()
+        hash_value = sha1(content).hexdigest()
+
         # Check actual hash against expected hash
+        if hash_value != a[0]:
+            break
+        return True
+        
         # Return False if any of the hashes do not match.
     return False
```

Comment:
The problem is that now, you return True if the first hash matches. You want to return True only if all hashes match, so I think you need something like the following:

```suggestion
        # Return False if any of the hashes do not match.
        if hash_value != a[0]:
            False
    return True
```

Reply:
Thanks a lot for pointing out the mistake, a very silly one on my part. I have accepted the minor suggestions and as for the mistake, I will first pull it and then change it on my local system and then push it again.